idaes/core/util/model_statistics.py

```diff
@@ -1034,6 +1060,19 @@ def unfixed_variables_in_activated_equalities_set(block):
     for v in variables_in_activated_equalities_set(block):
         if not v.fixed:
             var_set.add(v)
+
+    # Checks for greyboxes, and if they exist will add
+    # input and output vars to var_set if they are free
+    # inputs and outputs are defined names for greybox class and should always exist
+    for grey_box in _iter_indexed_block_data_objects(
+        block, ctype=ExternalGreyBoxBlock, active=True, descend_into=True
+    ):
+        for in_var in grey_box.inputs:
+            if not grey_box.inputs[in_var].fixed:
+                var_set.add(grey_box.inputs[in_var])
+        for out_var in grey_box.outputs:
+            if not grey_box.outputs[out_var].fixed:
+                var_set.add(grey_box.outputs[out_var])
```

Comment:
Here, I think you need to make sure that these variables don't already appear in `var_set`. If the `ExternalGreyBoxBlock` is constructed with the `inputs` or `outputs` argument, then the block will contain references to the user-provided variables, which may have already been counted.